src/Tic_Tac_toe.java

```diff
@@ -399,88 +401,69 @@ public static char Iswin(char[][] arr, char symbol) {
-        // TODO Auto-generated method stub
-        try {
-        	
-            FileWriter gameState = new FileWriter(filepath);
-            gameState.write(player + "\n");
-            gameState.write(Symbol1 + "\n");
-            gameState.write(Sympol2 + "\n");
-            for(int row = 0; row < arr.length; ++row) {
-                for(int col = 0; col < arr[0].length; ++col) {
-                    gameState.write(arr[row][col] + "\n");
-                }
-            }
-        } catch (IOException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
-    }
-	
+
+	public static void saveState(char[][] arr, String filepath) throws IOException {
+		PrintWriter writer = new PrintWriter(filepath);
+		for (int row = 0; row < arr.length; row++) {
+			for (int col = 0; col < arr.length; col++) {
+				writer.write(arr[row][col]);
+			}
+			if (row < 2)
+				writer.write("\n");
+			else
+				writer.write("\n");
+		}
+		writer.close();
+	} // end of the saveState method
+
 	/**
 	 * this method to chech the gmae state if the game endes.
+	 * 
 	 * @param filepath
 	 * @return
 	 */
 	private static boolean GameEnded(String filepath) {
-        // TODO Auto-generated method stub
-        File gameState = new File(filepath);
-        
-        if (gameState.exists()) {
-            return true;
-        } else {
-            return false;
-        }
-        
-        
-        
-        
-    }
-	
-	
+		// TODO Auto-generated method stub
+		File gameState = new File(filepath);
+
+		if (gameState.exists()) {
+			return true;
+		} else {
+			return false;
+		}
+
+	} // end of the GameEnded method
+
 	/**
 	 * method to load the last game so the player can resum the game.
+	 * 
+	 * @param arr      : the game board
 	 * @param filepath
-	 * @param arr : the game board
-	 * @param player : the player 1 or 2
-	 * @param Symbol1 : the symbol of player 1
-	 * @param Symbol2 : the symbol of player 2
+	 * @throws FileNotFoundException
 	 */
-	
-	private static void LastGame(String filepath, char[][] arr, int player, char Symbol1, char Symbol2) {
-        // TODO Auto-generated method stub
-        File gameState = new File(filepath);
-        
-        try {
-            Scanner stateScanner = new Scanner(gameState);
-            player = Integer.parseInt(stateScanner.nextLine());
-            Symbol1 = stateScanner.nextLine().charAt(0);
-            Symbol2 = stateScanner.nextLine().charAt(0);
-            
-            for(int row = 0; row < arr.length; ++row) {
-                for(int col = 0; col < arr[0].length; ++col) {
-                    arr[row][col] = stateScanner.nextLine().charAt(0);
-                }
-            }
-        } catch (FileNotFoundException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
-    }
-	
-	
-	
-
-	
+	public static void LastGame(char[][] arr, String filepath) throws FileNotFoundException {
```

Comment:
what do you mean by last game?

Reply:
Its a function that load the last exist game "Reading from file and display the board". So the player can resume the game

Reply:
Great